.github/workflows/ci.yml

```diff
@@ -0,0 +1,412 @@
+      PGHOST: postgres
+      PGUSER: postgres
+      PGPASSWORD: postgres
+
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v3
+      - name: Setup Python
+        uses: actions/setup-python@v3
+        with:
+          # FIXME: This fails with Python 3.10: "AttributeError: module
+          # 'importlib' has no attribute 'abc'. Did you mean: '_abc'?"
+          python-version: '3.9'
+      - name: Setup Rust
+        uses: ./.github/actions/setup-rust
+      - name: Set up environment
+        working-directory: tools/fog-local-network
+        run: |
+          python3 -m venv env
+          . ./env/bin/activate
+          pip install --upgrade pip
+          pip install -r requirements.txt
+          ./build.sh
+      - name: fog_conformance_tests.py
+        working-directory: tools/fog-local-network
+        run: |
+          . ./env/bin/activate
+          python3 fog_conformance_tests.py --release
+      - name: Upload core dumps
+        uses: ./.github/actions/upload-core-dumps
+      - name: Check dirty git
+        uses: ./.github/actions/check-dirty-git
+      - name: Upload logs
+        if: success() || failure()
+        uses: actions/upload-artifact@v3
+        with:
+          name: fog_conformance_tmp_fog
+          path: /tmp/fog*
+          if-no-files-found: ignore
+
+    services:
+      # Spin up a PostgreSQL server
+      postgres:
+        image: postgres
+        env:
+          POSTGRES_USER: postgres
+          POSTGRES_PASSWORD: postgres
+          POSTGRES_DB: postgres
+        # Set health checks to wait until postgres has started
+        options: >-
+          --health-cmd pg_isready
+          --health-interval 10s
+          --health-timeout 5s
+          --health-retries 5
+
+
+  fog-local-network-tests:
+    runs-on: [self-hosted, Linux, large]
+    container: gcr.io/mobilenode-211420/builder-install:1_28
+
+    env:
+      # DATABASE_URL points to a specific database.
+      DATABASE_URL: postgres://postgres:postgres@postgres/fog_local
+      # Set env vars so PG tools like createdb and dropdb do the right thing.
+      PGHOST: postgres
+      PGUSER: postgres
+      PGPASSWORD: postgres
+
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v3
+      - name: Setup Python
+        uses: actions/setup-python@v3
+        with:
+          python-version: '3'
+      - name: Setup Rust
+        uses: ./.github/actions/setup-rust
+      - name: Build and generate sample data
+        run: |
+          # Generate enclave signing key
+          openssl genrsa -out Enclave_private.pem -3 3072
+          export CONSENSUS_ENCLAVE_PRIVKEY="$PWD/Enclave_private.pem"
+          export INGEST_ENCLAVE_PRIVKEY="$PWD/Enclave_private.pem"
+          export LEDGER_ENCLAVE_PRIVKEY="$PWD/Enclave_private.pem"
+          export VIEW_ENCLAVE_PRIVKEY="$PWD/Enclave_private.pem"
+          export MC_LOG=debug
+
+          # Build binaries
+          cargo build \
```

Comment:
Might be worth to add a `--build-only` flag to fog-local-network, otherwise this list has to be kept in sync with whats in the script.